src/test/java/hu/dungeonhunter/TestFight.java

```diff
@@ -169,397 +179,302 @@ public void monsterCallerGoblinTest() {
-        MonstersInterface monster = monsterFactory.getMonster(CharacterTypes.GOBLIN);
+    public void fightInitiationCalculationMonsterTest() {
+        Goblin monster = new Goblin();
         monster.setHp(1);
-        monster.setAccuracy(1000);
-        monster.setInitiation(1000);
+        monster.setDefense(1);
+        monster.setInitiation(100);
         fight.setMonster(monster);
         champion.setHp(1);
+        champion.setDefense(1);
+        champion.setInitiation(10);
         fight.setChampion(champion);
-        fight.setLoopcounter(99);
-        fight.setMonsterFinalInitiation(1000);
-        fight.setChampionFinalInitiation(1000);
+        fight.setCharactersInBattle(Arrays.asList(monster, champion));
 
-        Assertions.assertThatExceptionOfType(RuntimeException.class).isThrownBy(() -> fight.battle());
+        fight.initiationCalc();
 
-        ifAssertThatIsChampionDefeated();
-        assertThat(fight.getLoopcounter()).isEqualTo(100);
+        assertThat(fight.getCharactersInBattle().get(0)).isEqualTo(monster);
     }
 
     @Test
-    public void battleMonsterWithHighInitiationAndHighAccuracyTest() {
-        MonstersInterface monster = monsterFactory.getMonster(CharacterTypes.GOBLIN);
-        monster.setAccuracy(1000);
-        fight.setMonster(monster);
-        fight.setMonsterFinalInitiation(1000);
-        lowHpChampion();
-
-        fight.battle();
-
-        ifAssertThatIsChampionDefeated();
-        ifAssertThatIsMonsterStillAlive();
-        softly.assertThat(fight.getMonsterFactory().getKilledMonsterCounter()).isEqualTo(startingKilledMonsterCounter);
-        softly.assertThat(fight.getMonsterCounter()).isEqualTo(startingMonsterCounter);
-        softly.assertThat(fight.getChampion().getHealingPotionCounter()).isEqualTo(startingHealingPotionCounter);
-        softly.assertThat(fight.nextTurn()).isFalse();
-        softly.assertAll();
+    public void getCharacterChampionTest() {
+        Assertions.assertThatExceptionOfType(RuntimeException.class).isThrownBy(() -> monsterFactory.getCharacter(CharacterTypes.CHAMPION));
     }
 
     @Test
-    public void battleMonsterWithLowHpHighInitiationAndChampWithHighDefenseAndAccuracyTest() {
-        MonstersInterface monster = monsterFactory.getMonster(CharacterTypes.GOBLIN);
-        monster.setHp(1);
+    public void monsterAccuracyIfAccuracyRollIs1Test() {
+        Goblin monster = new Goblin();
         fight.setMonster(monster);
-        fight.setMonsterFinalInitiation(1000);
-        Champion champion = new Champion();
-        champion.setHp(1 + fight.getMonster().getMaxDamage());
-        champion.setAccuracy(1000);
-        champion.setDefense(1000);
-        fight.setChampion(champion);
+        fight.setRandomCriticalMissEvent(5);
+        fight.attack(monster, 1);
 
-        fight.battle();
+        softly.assertThat(fight.getRandomCriticalMissEvent()).isLessThan(5);
```

Comment:
Don't use soft assertion if there is only a single assertion in the test.

Reply:
fixed